d1985r0.md

```diff
@@ -428,58 +427,88 @@ template<typename T, typename A> my_container(T*, size_t, const A&) -> my_contai
 
 
-## Overloading of variable templates
+## Impacts on the specialization of variable templates
 
-Universal template parameters can be used to implement what appears like overloading a variable
-template. The problem of not being able to delete the base case then becomes more
-pressing than currently as selecting the base case should trigger an error but
-will be unconditionally selected when none of the specialziations matches. The dependent
-static_assert mentioned above can be used to rescue this situation but
-this is still somewhat of a hack.
+Universal template parameters can be used to implement what appears
+like overloading of variable templates. 
+
+The problem of not being able to delete the base case then becomes
+more pressing than currently as selecting the base case should trigger
+an error but will be selected when none of the specialziations
+matches. This is solved by [@p2041r0] which is assumed in the example
+below.
 
 ```cpp
+
+// Metafunction to find a tuple element by a type predicate.
+template<template<typename> class PRED, size_t POS, typename TL> 
+constexpr size_t tuple_find()
+{
+    if constexpr (POS == tuple_size<TL>()) {
+        return npos;
+    }
+    else if constexpr (PRED<remove_cvref_t<tuple_element_t<POS, TL>>>::value)
+        return POS;
+    else
+        return tuple_find<PRED, POS + 1, TL>();
+}
+template<template<typename> class PRED, typename TL> 
+constexpr size_t tuple_find() { return tuple_find<PRED, 0, TL>(); }
+
+// Helper to bind the first arguments of a provided template
+template<template<typename auto...> TPL, template auto... BS> struct {
+    template<typename auto... Ts> using func = TPL<BS..., Ts>;
+};
+
 template<template auto... Ps>
-constexpr auto tuple_find_v = [] { static_assert(dependent_false_v<Ps...>); };
+constexpr auto tuple_find_v = delete;
 
 template<template<typename> class PRED, typename TL>
 constexpr size_t tuple_find_v<PRED, TL> = tuple_find<PRED, TL>();
 
+template<template<typename> class PRED, size_t POS, typename TL>
+constexpr size_t tuple_find_v<PRED, TL> = tuple_find<PRED, POS, TL>();
+
+// Convenience specialization for use with binary predicate
 template<template<typename, typename> class PRED, typename M, typename TL> 
-constexpr size_t tuple_find_v<PRED, M, TL> = tuple_find_v<metafunction_curry<PRED, M>, TL>;
+constexpr size_t tuple_find_v<PRED, M, TL> = tuple_find_v<curry<PRED, M>::template func, TL>;
 
+// Convenience specialization to match particular type.
 template<typename T, typename TL>
 constexpr size_t tuple_find_v<T, TL> = tuple_find_v<std::is_same, T, TL>;
 
 ```
 
 This example is a metafunction to find a matching element in a tuple
-based only on its type. The first overload is usable for a unary predicate like
-std::is_reference while the second overload is useful for a binary
-predicate like std::is_base_of by using a currying helper.
+based only on its type. Unfortunately today `tuple_find` can only be
```

Comment:
replace all "today"s etc with "in c++20" and so on. "Today", "now" etc are bad form for papers.

Reply:
true